### PR TITLE
Make "janusgraphmr.ioformat.filter-partitioned-vertices" option work

### PR DESCRIPTION
**Pull request by inho1213**
- "ConfigElement.getPath" in "AbstractConfiguration.getPath" returns "ioformat.filter-partitioned-vertices", but the "config" variable only contains "janusgraphmr.ioformat.filter-partitioned-vertices". So the "includeRoot" argument has been added to be able to include root path of the "ConfigOption".

Signed-off-by: inho1213 <inho1213@naver.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java [hunk 1]
```diff
@@ -32,48 +32,97 @@
...
+                return config.getSubset(umbrella, concat(umbrellaElements));
+            }
+
+            @Override
+            public Configuration restrictTo(String... umbrellaElements) {
+                return config.restrictTo(concat(umbrellaElements));
+            }
+        };
+    }
+
```
Comment (farodin91) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java]: Why did you move this up?

Comment (inho1213) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java]: Ah! It's my IDE formatting result. There is no reason.

Comment (farodin91) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java]: Could you revert this all the format changes make this pr hard to review

Comment (inho1213) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java]: OK. I did.

**Comment by farodin91** [comment]
@inho1213  Could you check why travis ci fails? It would be good to add a test. 

You can fix DCO bot before we merge the pr.

**Comment by inho1213** [comment]
@farodin91 

It seems to be done.

**Comment by farodin91** [comment]
@inho1213 Could you add an explicit test?

**Comment by inho1213** [comment]
> @inho1213 Could you add an explicit test?

IMHO, both 'CassandraScanJobIT' and 'CQLScanJobIT' already have showed that "janusgraphmr.ioformat.filter-partitioned-vertices" option works correctly.
Because 'HadoopInputFormat' which uses 'JanusGraphHadoopSetupImpl' is inherited by both 'CassandraInputFormat' and 'CQLInputFormat'.
Still need an explicit test?

**Comment by farodin91** [comment]
I thought more of a unit test for this method.

**Comment by inho1213** [comment]
> I thought more of a unit test for this method.

What method do you mean?
Test code for 'JanusGraphHadoopSetupImpl' or 'BasicConfiguration'?

**Comment by farodin91** [comment]
> > I thought more of a unit test for this method.
> 
> What method do you mean?
> Test code for 'JanusGraphHadoopSetupImpl' or 'BasicConfiguration'?

For BasicConfiguration

**Comment by inho1213** [comment]
I have committed some test code.
Does it fit your intend?

File: janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package org.janusgraph.hadoop;
```
Comment (farodin91) [inline janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java]: You have to add the copyright header.
I would move the test into following class: janusgraph/janusgraph-test/src/test/java/org/janusgraph/diskstorage/configuration/ConfigurationTest.java

**Comment by farodin91** [comment]
If you done with my comments could squash all commits and rebase on top of master.

File: janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java (hunk 2 above)
Comment (inho1213) [reply janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java]: Hm.. 'janusgraph-test' project has no hadoop dependency.
Also 'JanusGraphHadoopSetupImpl' uses 'ModifiableHadoopConfiguration' instead of 'ModifiableConfiguration' or 'BasicConfiguration'.
So I added test code to 'janusgraph-hadoop'.
Anyway do I modify test code using 'ModifiableConfiguration' or 'BasicConfiguration'?

**Comment by inho1213** [comment]
I don't know why 'graphConfigurationShouldBeWhatWeExpectWhenUsingTemplateConfiguration' failed.
In my local environment, there is no any test failure.
Is this related my code?

**Comment by FlorianHockmann** [comment]
> I don't know why 'graphConfigurationShouldBeWhatWeExpectWhenUsingTemplateConfiguration' failed.
In my local environment, there is no any test failure.
Is this related my code?

This is unfortunately a known flaky test (see #1495). I restarted the Travis job. Hopefully it will pass in the second run.

File: janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java (hunk 2 above)
Comment (farodin91) [reply janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java]: I'm okay with the current implementation. I thought it is independent from Hadoop because there are no changes to ModifiableHadoopConfiguration which extends ModifiableConfiguration. I think this part of the test should in the core test otherwise every change in configuration part has to be tested with Hadoop from now.

**Comment by inho1213** [comment]
It seems that all tests are passed when I see the detail link.
But the change request has not been changed to OK.
Is there any action remained to me?

**Comment by farodin91** [comment]
> @inho1213 I can you squash and rebase you commits. Afterwards, i will merge this PR.

Are you able to squash and rebase you commits?

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4283685&bid=14611507)

**Comment by inho1213** [comment]
> > @inho1213 I can you squash and rebase you commits. Afterwards, i will merge this PR.
> 
> Are you able to squash and rebase you commits?

I have finished rebase.

**Comment by porunov** [comment]
@inho1213 Could you please squash you 13 commits into a single commit? If you have problems with it, tell, I will help you squishing them

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4283685&bid=14611507)

**Comment by inho1213** [comment]
> @inho1213 Could you please squash you 13 commits into a single commit? If you have problems with it, tell, I will help you squishing them

Sorry to bother you guys. 
It seems to be done.

**Comment by farodin91** [comment]
@porunov do you want to review?

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- janusgraph-hadoop/src/test/java/org/janusgraph/hadoop/ConfigurationTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4283685&bid=14611507)

**Comment by inho1213** [comment]
@porunov 
Sorry for late. I have resolved conflict.
